scripts/z_sweep_jobs_submit.py

```diff
@@ -0,0 +1,132 @@
+
+1) The parameter file (required) is a tab separated file. The columns of the
+file indicate number of dimensions to constrain the model to learning. The
+index of the file indicate separate hyperparameters required as input to
+`train_models_given_z.py`. The parameters include:
+
+vae_epochs - number of epochs for Tybalt model
+dae_epochs - number of epochs for ADAGE model
+vae_lr - learning rate (step size) for Tybalt model
+dae_lr - learning rate (step size) for ADAGE model
+vae_batch_size - batch size for Tybalt model
+dae_batch_size - batch size for ADAGE model
+dae_noise - noise injected into ADAGE model training
+dae_sparsity - sparsity penalty regularization for ADAGE model training
+vae_kappa - warm up parameter for Tybalt model
+
+2) The tab sep config file has parameters for training on PMACS with a header:
+header = ["variable", "assign"]
+The values include:
+queue - the queue that will schedule and run jobs
+num_gpus - how many GPUs to request (if "gpu" queue requested)
+num_gpus_shared - how many of these GPUs run concurrent jobs
+walltime - how long to request for each job to run
+
+NOTE: Either a PMACS configuration file or a `--local` flag must be provided.
+
+Usage: Run in command line: python scripts/latent_space_sweep_submit.py
+
+     with required command arguments:
+
+       --param_config       location of tsv file (param by z dimension) for the
+                            specific parameter combination for each z dimension
+       --out_dir            filepath of where to save the results
+
+     and optional arguments:
+
+       --pmacs_config       filepath pointing to PMACS configuration file
+       --python_path        absolute path of PMACS python in select environment
+                              default: '~/.conda/envs/tybalt-gpu/bin/python'
+       --num_seeds          how many models to build (random seeds to set)
+                              default: 5
+       --components         a comma separated string of z dimensions
+       --local              if provided, sweep will be run locally instead
+
+Output:
+Submit jobs given certain number of latent space dimensionality
+"""
+
+import os
+import argparse
+import pandas as pd
+from bsub_helper import bsub_help
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-y', '--param_config',
+                    help='locaiton of the parameter configuration')
+parser.add_argument('-d', '--out_dir', help='folder to store results')
+parser.add_argument('-c', '--pmacs_config', default='config/pmacs_config.tsv',
+                    help='location of the configuration file for PMACS')
+parser.add_argument('-p', '--python_path', help='absolute path of python',
+                    default='python')
+parser.add_argument('-s', '--num_seeds', default=5,
+                    help='number of different seeds to run on current data')
+parser.add_argument('-n', '--components', help='dimensionality to sweep over',
```

Comment:
change to `nargs='+'`, so instead of having to write each number with a comma they can just put a space instead. Also instead of providing a vast list of components just make the command required.